mongo_cache_test.go

```diff
@@ -0,0 +1,298 @@
+package cache
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+var (
+	defSes = &mgo.Session{}
+	// session is the default session with default options
+	session = initMongo()
+
+	// config options for MongoCache
+	ttl = func(m *MongoCache) {
+		m.TTL = 2 * time.Minute
+	}
+
+	collection = func(m *MongoCache) {
+		m.CollectionName = "TestCollectionName"
+	}
+
+	gcInterval = func(m *MongoCache) {
+		m.GCInterval = 2 * time.Minute
+	}
+
+	startGC = func(m *MongoCache) {
+		m.GCStart = true
+	}
+)
+
+func TestMongoCacheConfig(t *testing.T) {
+	mgoCache := NewMongoCacheWithTTL(session)
+	if mgoCache == nil {
+		t.Fatal("config should not be nil")
+	}
+	configTTL := NewMongoCacheWithTTL(session, ttl)
+	if configTTL == nil {
+		t.Fatal("ttl config should not be nil")
+	}
+	if configTTL.TTL != time.Minute*2 {
+		t.Fatal("config ttl time should equal 2 minutes")
+	}
+	config := NewMongoCacheWithTTL(session, collection, startGC)
+	if config == nil {
+		t.Fatal("config should not be nil")
+	}
+	if config.CollectionName != "TestCollectionName" {
+		t.Fatal("config collection name should equal 'TestCollectionName'")
+	}
+	if config.GCStart != true {
+		t.Fatal("config StartGC option should not be true")
+	}
+}
+
+func TestMongoCacheSetOptionFuncs(t *testing.T) {
+	mgoCache := NewMongoCacheWithTTL(session)
+	if mgoCache == nil {
+		t.Fatal("config should not be nil")
+	}
+
+	duration := time.Minute * 3
+	configTTL := NewMongoCacheWithTTL(session, SetTTL(duration))
+	if configTTL == nil {
+		t.Fatal("ttl config should not be nil")
+	}
+	if configTTL.TTL != duration {
+		t.Fatal("config ttl time should equal 2 minutes")
+	}
+
+	// check multiple options
+	collName := "testingCollectionName"
+	config := NewMongoCacheWithTTL(session, SetCollectionName(collName), SetGCInterval(duration), StartGC())
+	if config == nil {
+		t.Fatal("config should not be nil")
+	}
+	if config.CollectionName != collName {
+		t.Fatal("config collection name should equal 'TestCollectionName'")
```

Comment:
message is wrong, you can use the collName as parameter here instead of writing it again with a typo